std/experimental/logger/core.d

```diff
@@ -1631,7 +1631,7 @@ private @property Logger defaultSharedLoggerImpl() @trusted
     import std.conv : emplace;
     import std.stdio : stderr;
 
-    static __gshared ubyte[__traits(classInstanceSize, FileLogger)] _buffer;
+    static __gshared align(FileLogger.alignof) void[__traits(classInstanceSize, FileLogger)] _buffer;
```

Comment:
@andralex druntime & phobos are full of such ad hoc  implementations of the singleton behavior (the last being [5355](https://github.com/dlang/phobos/pull/5355/files#diff-0002a05f29875f98fa52cfba8996ac49R547)) we should really try to get `singleton!T` merged and fix this once and for all. Do you want me to take over the singleton implementation from https://github.com/dlang/druntime/pull/1710?